server/test/lib/scheduler/Schedule.spec.ts

```diff
@@ -0,0 +1,238 @@
+import * as chai from "chai";
+const { expect } = chai;
+import * as sinon from "sinon";
+
+import { Schedule } from "../../../lib/scheduler/Schedule";
+
+import {
+  USER_ID,
+  USER_ID_1,
+  USER_ID_2,
+  WORKSPACE_ID,
+  WORKSPACE_ID_1,
+  WORKSPACE_ID_2,
+  WORKSPACE_ID_3,
+  rootParentFinderFake,
+} from "./utils";
+
+const ONE_MINUTE = 60 * 1000;
+
+describe("Schedule class", () => {
+  let fakeClock, schedule;
+  beforeEach(() => {
+    fakeClock = sinon.useFakeTimers();
+    schedule = new Schedule({
+      rootParentFinder: rootParentFinderFake,
+      timeLimit: ONE_MINUTE,
+    });
+  });
+
+  it("has tests that work", () => {
+    expect(true).to.equal(true);
+  });
+
+  // tests private method, maybe remove
+  describe("doesUserHaveASchedule", () => {
+    it("works when user doesn't have a schedule", () => {
```

Comment:
I suggest using ``contexts`` as a second layer to make this a bit more specific, a la Ruby.

http://www.albertgao.xyz/2016/12/09/testing-via-mocha-and-chai-in-es5-and-es6/